### PR TITLE
Samples table fixed header and footer

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Fixed some contrast issues in dark mode on samples table.

Refactored samples listing to have a scrollable table with fixed header and footer, which shrinks so that page does not take more than viewport. Also customized scrollbar to display properly in safari (style also applies to Chrome, but not firefox).

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Launch server
2. Go to a Group with lots of samples
3. Toggle metadata then change page size to 100
4. Verify that page doesn't scroll, and that the table scrolls horizontally vertically.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/components/samples/table_component.html.erb [hunk 1]
```diff
@@ -9,7 +9,7 @@
       <thead
         class='
           text-xs uppercase text-slate-700 bg-slate-50 dark:bg-slate-700
-          dark:text-slate-400
+          dark:text-slate-300 sticky top-0 z-10 special-thead
```
Comment (joshsadam) [inline app/components/samples/table_component.html.erb:12]: I don't think you are defining the class `special-thead` anywhere

File: app/assets/stylesheets/application.tailwind.css [hunk 2]
```diff
@@ -478,3 +478,25 @@ div.field_with_errors > :is(select) {
 .table-container .empty_state_message {
   display: none;
 }
+
+.samples-app {
```
Comment (joshsadam) [inline app/assets/stylesheets/application.tailwind.css:482]: Shouldn't this be something more like `scrollable-wrapper` or something?  We might want to use this on a non-samples table.

**Review by joshsadam — CHANGES_REQUESTED** [review]
This looks really good.  Just a couple suggestions.  I would definitely like to see (at a later point) keeping page size and scroll when toggling metadata.

File: app/assets/stylesheets/application.tailwind.css (hunk 2 above)
Comment (ericenns) [reply app/assets/stylesheets/application.tailwind.css:482]: This actually non-scrollable or at least intended to be non-scrollable. E.G. wrapper that takes up remaining browser size. Not sure what that should be named?

**Review by ChrisHuynh333 — APPROVED** [review]
Looks good!

File: app/components/samples/table_component.html.erb (hunk 1 above)
Comment (ericenns) [reply app/components/samples/table_component.html.erb]: Removed in https://github.com/phac-nml/irida-next/pull/725/commits/9dede8b63be4018bea710a82ec1afc560feabf16

**Review by joshsadam — APPROVED** [review]

**Pull request merged**